src/form/GeometryField.tsx

```diff
@@ -0,0 +1,141 @@
+import React, {useState, useEffect} from 'react';
+import {FloatInput} from './FloatInput';
+import {
+  AssetLocationValue
+} from "../types/locationFormTypes";
+import formStyles from "../styles/Forms.module.css";
+
+ 
+
+interface MyProps {
+  title: string,
+  name: string,
+  value: AssetLocationValue,
+  validated: boolean,
+  valueChanged: (value: AssetLocationValue) => void,
+  clearInput?: (e: any) => void,
+  errorMessage?: string | false,
+  placeholder?: string,
+  handleEnter?: (e: any) => void,
+  onFocus?: (e: React.ChangeEvent<HTMLInputElement>) => void,
+  onBlur?: () => void,
+  triedToSubmit?: boolean,
+  readOnly?: boolean
+};
+
+export const GeometryField: React.FC<MyProps> = (props) => {  
+  const {
+    title,
+    name,
+    placeholder,
+    value,
+    validated,
+    valueChanged,
+    onFocus,
+    onBlur,
+    handleEnter,
+    clearInput,
+    errorMessage,
+    triedToSubmit,
+    readOnly
+  } = props;
+
+  const [lat, setLat] = useState(NaN);
+  const [lng, setLng] = useState(NaN);
+
+
+  useEffect(() => {
+    if ( value.asset && value.asset.value && value.asset.value.view ) {
+      setLat(value.asset.value.view[0]);
+      setLng(value.asset.value.view[1])
+    } else if (value.location) {
+      setLat(value.location.lat);
+      setLng(value.location.lng); 
+    } else {
+      setLat(NaN);
+      setLng(NaN);
+    }
+  }, [value.asset, value.location]);
```

Comment:
I am not sure if this field can be reused by the RasterAlarmForm? Maybe with some tweaks it would be possible.